Quarkus CLI report error when using CLI plugins with flags

### Describe the bug

If I add an extension that provides a CLI plugin and try to check its usage. For example:

```shell
quarkus helm --help
```

It reports:

```
Command quarkus-helm is not available, looking for available plugins ...
[ERROR] ❗  Command quarkus-helm is missing and can't be installed.
Helm CLI
Usage: helm [-h] [COMMAND]
  -h, --help   Display this help message.
Commands:
  install    Helm Install
  upgrade    Helm Upgrade
  uninstall  Helm Uninstall
  list       List Helm charts in the project
  lint       Helm Lint
```

The command does run properly despite an error being reported.

### Expected behavior

```
Helm CLI
Usage: helm [-h] [COMMAND]
  -h, --help   Display this help message.
Commands:
  install    Helm Install
  upgrade    Helm Upgrade
  uninstall  Helm Uninstall
  list       List Helm charts in the project
  lint       Helm Lint
```

### Actual behavior

```
Command quarkus-helm is not available, looking for available plugins ...
[ERROR] ❗  Command quarkus-helm is missing and can't be installed.
Helm CLI
Usage: helm [-h] [COMMAND]
  -h, --help   Display this help message.
Commands:
  install    Helm Install
  upgrade    Helm Upgrade
  uninstall  Helm Uninstall
  list       List Helm charts in the project
  lint       Helm Lint
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/cli/src/main/java/io/quarkus/cli/QuarkusCli.java
+++ b/devtools/cli/src/main/java/io/quarkus/cli/QuarkusCli.java
@@ -12,6 +12,7 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.Callable;
 import java.util.function.Supplier;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 import jakarta.inject.Inject;
@@ -176,7 +177,8 @@ public class QuarkusCli implements QuarkusApplication, Callable<Integer> {
                         .collect(Collectors.toList());
                 if (!unmatchedSubcommands.isEmpty()) {
                     missingCommand.append("-").append(unmatchedSubcommands.get(0));
-                    return Optional.of(missingCommand.toString());
+                    // We don't want the root itself to be added to the result
+                    return Optional.of(missingCommand.toString().replaceFirst(Pattern.quote(root.getCommandName() + "-"), ""));
                 }
 
                 currentParseResult = currentParseResult.subcommand();

--- a/devtools/cli/src/main/java/io/quarkus/cli/QuarkusCli.java
+++ b/devtools/cli/src/main/java/io/quarkus/cli/QuarkusCli.java
@@ -12,7 +12,6 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.Callable;
 import java.util.function.Supplier;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 import jakarta.inject.Inject;
@@ -178,7 +177,7 @@ public class QuarkusCli implements QuarkusApplication, Callable<Integer> {
                 if (!unmatchedSubcommands.isEmpty()) {
                     missingCommand.append("-").append(unmatchedSubcommands.get(0));
                     // We don't want the root itself to be added to the result
-                    return Optional.of(missingCommand.toString().replaceFirst(Pattern.quote(root.getCommandName() + "-"), ""));
+                    return Optional.of(stripRootPrefix(missingCommand.toString(), root.getCommandName() + "-"));
                 }
 
                 currentParseResult = currentParseResult.subcommand();
@@ -193,6 +192,14 @@ public class QuarkusCli implements QuarkusApplication, Callable<Integer> {
             // This will be handled by Picocli at a later step.
             return Optional.empty();
         }
+    }
+
+    private static String stripRootPrefix(String command, String rootPrefix) {
+        if (!command.startsWith(rootPrefix)) {
+            return command;
+        }
+
+        return command.substring(rootPrefix.length());
     }
 
     @Override

